remix.init/index.js

```diff
@@ -1,75 +1,72 @@
-
-// Netlify Edge Functions template file changes
-const edgeFilesToCopy = [
-  ["README-edge.md", "README.md"],
-  ["netlify-edge-toml", "netlify.toml"],
-  ["server.ts"],
-  ["remix.config.js"],
-  ["entry.server.tsx", "app/entry.server.tsx"],
-  ["root.tsx", "app/root.tsx"],
-  ["vscode.json", ".vscode/settings.json"],
-];
-
-// Netlify Functions template file changes
-const filesToCopy = [["README.md"], ["netlify-toml", "netlify.toml"]];
-
-async function copyTemplateFiles({ files, rootDirectory }) {
-  for (const [file, target] of files) {
-    let sourceFile = file;
-    let targetFile = target || file;
-
-    await fs.copyFile(
-      join(rootDirectory, "remix.init", sourceFile),
-      join(rootDirectory, targetFile)
-    );
+
+const foldersToRemove = [".github"];
+
+const packagesToRemove = {
+  edge: ["@netlify/remix-adapter", "@netlify/functions"],
+  functions: [
+    "@netlify/edge-functions",
+    "@netlify/remix-edge-adapter",
+    "@netlify/remix-runtime",
+  ],
+};
+
+async function mergeDirs(src, dest) {
+  await fs.mkdir(dest, { recursive: true });
+  let entries = await fs.readdir(src, { withFileTypes: true });
+
+  for (let entry of entries) {
+    const srcPath = join(src, entry.name);
+    const destPath = join(dest, entry.name);
+
+    if (entry.isDirectory()) {
+      await mergeDirs(srcPath, destPath);
+    } else {
+      await fs.mkdir(dirname(destPath), { recursive: true });
+      await fs.copyFile(srcPath, destPath);
+    }
   }
 }
 
-async function updatePackageJsonForEdge(directory) {
-  const packageJson = await PackageJson.load(directory);
-  const {
-    dependencies: { "@remix-run/node": _node, ...dependencies },
-    scripts,
-    ...restOfPackageJson
-  } = packageJson.content;
-
-  packageJson.update({
-    // dev script is the same as the start script for Netlify Edge, "cross-env NODE_ENV=production netlify dev"
-    scripts: {
-      ...scripts,
-      dev: 'remix dev --manual -c "ntl dev --framework=#static"',
-    },
-    ...restOfPackageJson,
-    dependencies: {
-      ...dependencies,
-      "@netlify/edge-functions": "^2.0.0",
-      "@netlify/remix-edge-adapter": "^3.0.0",
-      "@netlify/remix-runtime": "^2.0.0",
-    },
-  });
-
-  await packageJson.save();
+async function copyTemplateFiles(rootDirectory, useEdge) {
+  const source = join(
+    rootDirectory,
+    "remix.init",
+    useEdge ? "edge" : "functions"
+  );
+  await mergeDirs(source, rootDirectory);
 }
 
-async function updatePackageJsonForFunctions(directory) {
+async function removeUpdatePackageJson(directory, dependencies) {
   const packageJson = await PackageJson.load(directory);
-  const { dependencies, ...restOfPackageJson } = packageJson.content;
+
+  const { dependencies: currentDependencies = {}, scripts } =
+    packageJson.content;
+
+  // Remove the auto-init command from the scripts
+  for (const script of ["build", "start", "dev"]) {
+    scripts[script] = scripts[script]?.replace("remix init && ", "");
```

Comment:
@pieh here

Reply:
should we have concerns for mutating `package.json` and possibly causing surprising modifications and git tracking?

Reply:
We need to mutate it anyway, because they each have different dependencies. In almost all cases this is run as part of the template setup from `create-remix`, which commits the changed files. Even in the case where it's run from the package.json command there'll still be a stack of modifications anyway because we're moving all the files around to setup the template.

Reply:
I was mostly thinking about cases when it wouldn't run at init for some reason (I did assume that template init does handle commiting initial modifications), but in this case there's not much for us to do, possibly than try to recognize it's not init execution and maybe print message / try to commit ourselves (but this is can of worms and probably leaving user with modified files and letting them handle it is better than trying to handle that ourselves)